tests/gtfs/test_gtfs_utils.py

```diff
@@ -40,3 +47,57 @@ def test_bbox_filter_gtfs_writes_as_expected(self, tmpdir):
         assert isinstance(
             feed, GtfsInstance
         ), f"Expected class `Gtfs_Instance but found: {type(feed)}`"
+
+
+class TestConvertPandasToPlotly(object):
+    """Test convert_pandas_to_plotly()."""
+
+    def test_convert_pandas_to_plotly_defences(self):
+        """Test convert_pandas_to_plotly defences."""
+        test_df = pd.DataFrame(
+            {
+                "ID": [1, 2, 3, 4, 1],
+                "score": [45, 34, 23, 12, 23],
+                "grade": ["A", "B", "C", "D", "C"],
+            }
+        )
+        with pytest.raises(
+            LookupError,
+            match=re.escape(
+                "dark is not a valid colour scheme."
+                "Valid colour schemes include ['dsc']"
+            ),
+        ):
+            convert_pandas_to_plotly(test_df, scheme="dark")
+
+        multi_index_df = test_df.groupby(["ID", "grade"]).agg(
+            {"score": ["mean", "min", "max"]}
+        )
+        with pytest.raises(
+            TypeError,
+            match="Pandas dataframe must have a single index,"
+            "not MultiIndex",
+        ):
+            convert_pandas_to_plotly(multi_index_df)
+
+    def test_convert_pandas_to_plotly_on_pass(self):
+        """Test convert_pandas_to_plotly() when defences pass."""
+        test_df = pd.DataFrame(
+            {
+                "ID": [1, 2, 3, 4, 1],
+                "score": [45, 34, 23, 12, 23],
+                "grade": ["A", "B", "C", "D", "C"],
+            }
+        )
```

Comment:
As this is repeating code from above, a locally scoped fixture will do the trick here. Like how you handled the fixture in tests/gtfs/report/test_report_utils.py `@pytest.fixture(scope="function")`